src/pages/settings/Subscription/RequestEarlyCancellationPage/index.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import Button from '@components/Button';
+import FeedbackSurvey from '@components/FeedbackSurvey';
+import FixedFooter from '@components/FixedFooter';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as Report from '@userActions/Report';
+import type {CancellationType, FeedbackSurveyOptionID} from '@src/CONST';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+
+function RequestEarlyCancellationPage() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    // TODO: replace this with NVP_PRIVATE_CANCELLATION_DETAILS.cancellationType
+    const [cancellationType, setCancellationType] = useState<CancellationType>();
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const handleSubmit = (cancellationReason: FeedbackSurveyOptionID, cancellationNote = '') => {
+        // TODO: call the CancelBillingSubscriptionNewDot API method
+        setCancellationType(CONST.CANCELLATION_TYPE.AUTOMATIC);
+    };
+
+    const acknowledgmentText = (
+        <Text>
+            {translate('subscription.requestEarlyCancellation.acknowledgement.part1')}
+            <TextLink href={CONST.TERMS_URL}>{translate('subscription.requestEarlyCancellation.acknowledgement.link')}</TextLink>
+            {translate('subscription.requestEarlyCancellation.acknowledgement.part2')}
+        </Text>
+    );
+
+    let screenContent: React.ReactNode;
+
+    if (cancellationType === CONST.CANCELLATION_TYPE.MANUAL) {
```

Comment:
This is becoming a bit difficult to follow. Do you think we could split them into individual `useMemo` based render components and replace this with `switch..case`